### PR TITLE
Enhanced the WikipediaPipelineApp to accept an application config. Th…

### PR DESCRIPTION
**Pull request by bdmogal**
…e app config allows users to choose the clustering algorithm that will be used to analyze wikipedia data. Added a K-Means algorithm from Apache Spark in addition to the existing LDA algorithm to demonstrate this use case. Re-factored the Scala code to allow for re-usability between the two algorithms. Also added two application config jsons for choosing either of the two algorithms.

Build: http://builds.cask.co/browse/CDAP-DUT3021

### CODE REVIEW

**Comment by bdmogal** [comment]
Mostly moving around the scala code from ``ScalaSparkLDA`` into ``ClusteringUtils``, so it can be shared with ``ScalaSparkKMeans``. Only other major change is that the app now accepts a ``Config``, that is used to determine the program to run in the ``ScalaSparkClustering`` node, and also set the Spark program's name based on the chosen algorithm.

**Comment by awholegunch** [comment]
Fails [Quick Build](http://builds.cask.co/browse/CDAP-DOB32-1). It's because develop is broken. Changes by @chtyim were not updated in the doc build, and so it is currently broken.
I've made edits; please check them. Docs otherwise LGTM.

**Comment by bdmogal** [comment]
Thanks @awholegunch.

File: cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst [hunk 1]
```diff
@@ -88,6 +96,28 @@ This application demonstrates:
...
 
 
+Deploying the Example
+=====================
+Since deploying the WikipediaPipelineApp involves loading an artifact and creating an application from it, the
+preferred method of deploying it is to use the CDAP CLI.
+
+- Load the Artifact::
+
+    $ cdap-cli.sh load artifact target/WikipediaPipeline-3.3.0-SNAPSHOT.jar name WikipediaPipelineApp version
```
Comment (albertshau) [inline cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst]: does this command work? Looks like the command takes config-file before name and version, not sure if CLI does varargs type of args.  You could also leave out name and version and let the command derive them from the filename.

Comment (bdmogal) [inline cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst]: It does work. The ``config-file`` param seems optional. In this case, I do not need a config that specifies parents etc.

```
cdap (http://localhost:10000/default)> load artifact ~/work/git/cdap/cdap-examples/WikipediaPipeline/target/WikipediaPipeline-3.3.0-SNAPSHOT.jar name WikipediaPipelineApp version 3.3.0-SNAPSHOT
Successfully added artifact with name 'WikipediaPipelineApp'

cdap (http://localhost:10000/default)> load artifact ~/work/git/cdap/cdap-examples/WikipediaPipeline/target/WikipediaPipeline-3.3.0-SNAPSHOT.jar
Successfully added artifact with name 'WikipediaPipeline'

cdap (http://localhost:10000/default)> load artifact ~/work/git/cdap/cdap-examples/WikipediaPipeline/target/WikipediaPipeline-3.3.0-SNAPSHOT.jar name WikipediaA version 3.4.0-SNAPSHOT
Successfully added artifact with name 'WikipediaA'

cdap (http://localhost:10000/default)> list artifacts user
+===============================================+
| name                 | version        | scope |
+===============================================+
| WikipediaA           | 3.4.0-SNAPSHOT | USER  |
| WikipediaPipeline    | 3.3.0-SNAPSHOT | USER  |
| WikipediaPipelineApp | 3.3.0-SNAPSHOT | USER  |
+===============================================+
```

Comment (albertshau) [inline cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst]: cool, good to know

**Comment by albertshau** [comment]
lgtm

**Comment by bdmogal** [comment]
Build passed - http://builds.cask.co/browse/CDAP-DUT3021-5. Merging.

**Pull request merged**